Application/components/FriendsColumn/AddFriendModal.tsx

```diff
@@ -1,15 +1,25 @@
 import { useState, useEffect } from 'react';
 import { useDisclosure } from '@mantine/hooks';
-import { Modal, Text, Stack, Button, TextInput } from '@mantine/core';
-import { NewFriendModalProps } from '@/accordTypes'; // Make sure this path is correct for your project
-import { showNotification } from '@mantine/notifications';
+import { Modal, Tooltip, ActionIcon, Text, Stack, Button, TextInput, useMantineTheme } from '@mantine/core';
+import { IconPlus } from '@tabler/icons-react';
+import { NewFriendModalProps } from '@/accordTypes';
+import { notifications, showNotification } from '@mantine/notifications';
 
 export function AddFriendModal({ senderID, setLastFetched }: NewFriendModalProps) {
   const [opened, { open, close }] = useDisclosure(false);
   const [friendUsername, setFriendUsername] = useState('');
   const [searchResult, setSearchResult] = useState<number | null>(null);
   const [errorMessage, setErrorMessage] = useState('');
 
+  const theme = useMantineTheme();
+
+    /**
+   * Handles the logic for sending a friend request based on the entered username. It performs a POST
+   * request to the server with the sender's ID and the friend's username. Upon a successful request,
+   * it resets the input field for further requests and triggers a re-fetch of the friend list by updating
+   * the `lastFetched` state. If the request fails or if no username is entered, it sets an appropriate
+   * error message.
+   */
   const handleAddFriendClick = async () => {
     if (friendUsername.trim()) {
       try {
```

Comment:
- Ensure that error messages are displayed to the user in a clear and concise manner. Currently, the error handling logic seems incomplete, and the error message is not consistently displayed to the user. Consider refining the error handling mechanism to provide better feedback to users when friend requests fail or encounter errors.